### DISCUSSION TITLE
Diseño "responsive" (es imposible?)

### DISCUSSION THREAD

**Discussion by martinmurillo2020** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#296] [Q&A]
Holas a todos! vengo trabajando arduas horas en la clonación de la pagina de vacunación y de repente me voy cuenta que solo se ve bien en mi pantalla y con el zoom en 100%, es decir, siento que trabajé mas de 15 horas (y solo pude hacer el header y la primer imagen) entonces me doy cuenta que para pensar el diseño y las ubicaciones me falta mucho conocimiento. 

nunca se cuando usar px, %, em, etc... y supongo que eso tiene que ver a la hora de hacer responsive el diseño.

tampoco se como se hace eso, vi por ahí que usan un arroba al final y como que ahí ajustan todo. pero creo que como venga haciendo todo sin pensar en esto, voy a seguir trabajando en esto sin ningún sentido ya que a la hora de hacerlo responsive, voy a tener un trabajo igual de pesado y complicado o bien ni siquiera pueda hacerlo ya que tendría que modificar los parámetros principales antes.

me explico o estoy muy perdido???

**Comment by martinmurillo2020** [comment]
Así se ve cuando la tengo en 100% al zoom.
![image](https://user-images.githubusercontent.com/99059543/157107107-2bdd6ab6-b6f8-44f1-9977-7681792bbbbb.png)

**Comment by martinmurillo2020** [comment]
y cuando apenas reduzco la pantalla:

![image](https://user-images.githubusercontent.com/99059543/157107397-e958b41d-12e9-496d-b2db-52c0fe67fbe3.png)

**Comment by manusanabria** [comment]
Revisa el tema de media queries, para el diseño responsivo.

[https://developer.mozilla.org/es/docs/Web/CSS/Media_Queries/Using_media_queries](url)

**Reply by martinmurillo2020** [reply]
no funciona la pag.

**Reply by manusanabria** [reply]
Si funciona pero necesitas copiar y pegar el enlace en la barra de direcciones.

**Reply by martinmurillo2020** [reply]
dale, gracias!

**Reply by SujeiRodriguez** [reply]
yo también quería saber eso... :D gracias

**Comment by doguedogue** [comment]
Yo igual batallé con el responsive en la pastelería, usando un contenedor main después del body, se puede empezar haciendo que la vista tenga lectura de toda la pantalla.

Y como dice Manuel @manusanabria  los _media queries_

Además uso la herramienta de desarrollador en chrome y puedes ver tu página con dimensiones de celular.
![image](https://user-images.githubusercontent.com/23409026/157112205-a888cd40-f861-4194-81d8-5ecc6e78054a.png)



```
* {
	box-sizing: border-box;
	margin: 0;
	padding: 0;
}

main {
	width: 100vw;
	height:100vh;
	background-color: blue;
}
```

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>HTML y CSS</title>
    <link rel="stylesheet" href="/css/style.css">
</head>
<body>
    <main>
        Hola mundo
    </main>
</body>
</html>
```

**Reply by martinmurillo2020** [reply]
si, la uso, pero no se como se hace, solo puedo ver lo mal que estoy haciendo pero no como solucionarlo hahah

**Reply by doguedogue** [reply]
https://www.youtube.com/watch?v=2BtdGYS5fQ4

**Comment by AngelCruzO** [comment]
Con flexbox se soluciona un poco el reaponsive, pero aún así se ocupan los media querys para poder darle los tamaños adecuados para otras pantallas

**Comment by JuanSalvador73** [comment]
Primero haces el diseño tal y como lo hiciste, luego agregas los mediasQuerys como el en video de doguedogue que te envió.  En realidad es muy sencillo, solo si acaso es práctica y familiarizarte con el tema.

**Comment by Ingrzs** [comment]
Estúdia flexbox te ahorra un montón de tiempo, solo ajustas el mediaquery y listo ya que al mediaquery solo condicionas flexbox en columna 😅

**Comment by carlos-gmx3** [comment]
Usa flexbox y grid según creas conveniente, pero lo que más te tira paro es el uso de media queries
Básicamente son grupos de estilos que aplican según las condiciones que tu le indiques

@media (condiciones) {

}

Recuerdas que aparte de la especificidad, si tu aplicas un estilo a la misma etiqueta 2 veces gana el último?
bueno pues así más o menos funciona, tu estableces unos estilos por default y unos cuando cambien sus condiciones
puedes acondicionar tu pagina para que la vista default sea en escritorio y establecer media queries para cuando la resolucion de la pantalla sea menor a x pixeles
.estilosEscritorio {

}
@media (si el tamaño de la pantalla es menor a 720 pixeles aplica estos estilos) {
.estilosMovil {

}
}
o viceversa, acondicionas tu pagina para que la vista default se movil y cuando la pantalla exceda ciertas dimensiones  apliquen otros estilos

**Comment by DanyVeneno** [comment]
No te desanimes Martin asi se aprende vas a tener que hacer un stop para entender en donde te estas atorando y buscar soluciones es imposible retener todo y para que sirve cada cosa mejor ve haciendo una lista de sitios de consulta te dejo unos sitios en donde yo me apoyo :
1-https://css-tricks.com/
2-https://www.youtube.com/watch?v=udGrXWeJp1Y&ab_channel=jonmircha
3-https://www.w3schools.com/
4-https://es.stackoverflow.com/
5-https://www.udemy.com/course/curso-de-css3-flexbox-y-css-grid-layout-basico/
en este ultimo create una cuenta y entra a este curso es gratis aunque con el de youtube esta bueno igual .
y no tengas miedo empezar de 0 ni las hora que le hayas invertido si es necesario empezar de cero no lo dudes hay un lema que dice que entra más rapido te equivoques más rapído obtienes las soluciones .
si tienes más dudas aqui andamos saludos buena tarde
